snforge_std/src/errors.cairo

```diff
@@ -0,0 +1,31 @@
+use core::option::OptionTrait;
+use core::starknet::SyscallResult;
+use core::byte_array::BYTE_ARRAY_MAGIC;
+
+pub enum PanicDataOrString {
+    String: ByteArray,
+    PanicData: Array<felt252>,
+}
+
+type ExtendedSyscallResult<T> = Result<T, PanicDataOrString>;
+
+trait SyscallResultStringErrorTrait<T> {
+    fn map_string_errors(self: SyscallResult<T>) -> ExtendedSyscallResult<T>;
```

Comment:
```suggestion
    fn map_string_error(self: SyscallResult<T>) -> ExtendedSyscallResult<T>;
```
There is always just a single error (if any)

Reply:
[Change name to singular error](https://github.com/foundry-rs/starknet-foundry/pull/1679/commits/74417e38351cd016b9e4e13eb70e8a3db76434b6)